scripts/ci-e2e.sh

```diff
@@ -33,8 +33,42 @@ source "${REPO_ROOT}/hack/ensure-kind.sh"
 source "${REPO_ROOT}/hack/ensure-kubectl.sh"
 # shellcheck source=../hack/ensure-kustomize.sh
 source "${REPO_ROOT}/hack/ensure-kustomize.sh"
+# shellcheck source=../hack/parse-prow-creds.sh
+source "${REPO_ROOT}/hack/parse-prow-creds.sh"
 
-export REGISTRY=e2e
+# Verify the required Environment Variables are present.
+: "${AZURE_SUBSCRIPTION_ID:?Environment variable empty or not defined.}"
+: "${AZURE_TENANT_ID:?Environment variable empty or not defined.}"
+: "${AZURE_CLIENT_ID:?Environment variable empty or not defined.}"
+: "${AZURE_CLIENT_SECRET:?Environment variable empty or not defined.}"
+
+get_random_region() {
+    local REGIONS=("eastus" "eastus2" "southcentralus" "westus2" "westeurope")
```

Comment:
Are these regions the prow sub will have quota for? If not will we override it to a specific region for the PR job?

Reply:
We should have enough quota since each region above has at least 200 cores available (note that this is a clone of https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/test/e2e/resource_generators.go#L462-L474)